### ISSUE TITLE
Fix request: PMD7 AvoidUnguardedAssignmentToNonFinalFieldsInObjectsUsingSynchronized false positive

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#385]
matches local var, should be class field matches only

```java
package com.jpinpoint.tryout;

import net.jcip.annotations.GuardedBy;

import java.util.HashMap;
import java.util.Map;

public class NoMatchOnNonFieldValues {
    @GuardedBy("this")
    private static Map staticMapField = new HashMap();

    public String getDetails() {

        String detailsResponse = null;
        try {
            detailsResponse = produceResponse(); // <-- false positive
        } catch (Exception e) {
            throw new RuntimeException(e.getMessage());
        }

        return detailsResponse;
    }

    public static synchronized Map getStaticMapField() {
        return staticMapField;
    }
}
```
